File: Creational-Patterns/C++/Builder/src/builder/director.cpp
#include "director.hpp"


void builder_pattern::Director::set_builder( Builder *builder_type )
{
    this->builder = builder_type;
}

operational_system::OperationalSystem * builder_pattern::Director::get_operational_system()
{
    hardware_infos::Processor *proc = this->builder->get_processor();
    hardware_infos::Memory *mem = this->builder->get_memory();
    hardware_infos::HardDisk *hd = this->builder->get_hd();

    operational_system::OperationalSystem *os = new operational_system::OperationalSystem( mem,proc, hd );
    return os;
}

File: Creational-Patterns/Python/Builder/builder_pattern/director.py
#coding: utf-8

from builder_pattern import builder
from op_system import operational_system

class Director(object):

    def set_builder(self, builder_type : builder.AbstractBuilder):
        self.__builder = builder_type

    def get_operational_system(self):
        return operational_system.OperationalSystem(self.__builder.get_processor(), self.__builder.get_memory(), self.__builder.get_harddisk())
